numba/tests/test_dyn_array.py

```diff
@@ -600,6 +600,7 @@ def func(arr):
             return pyfunc(arr)
         self.check_like(func, np.float64)
 
+    @unittest.skipIf(numpy_version <= (1, 6), "test fails on Numpy 1.6")
```

Comment:
I think the problem is not that this feature is unsupported in Numpy 1.6, but rather that the use of `ndarray.fill()` on a structured dtype array in `check_like()` isn't supported on Numpy 1.6.

From Jenkins:
```
======================================================================
ERROR: test_like_dtype_structured (numba.tests.test_dyn_array.TestNdEmptyLike)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/build/anaconda/envs/_test/lib/python2.7/site-packages/numba/tests/test_dyn_array.py", line 628, in test_like_dtype_structured
    self.check_like(func, np.float64)
  File "/home/build/anaconda/envs/_test/lib/python2.7/site-packages/numba/tests/test_dyn_array.py", line 584, in check_like
    ret.fill(123)
TypeError: expected a readable buffer object
```